esp-hal-common/src/analog/adc/riscv.rs

```diff
@@ -1,16 +1,71 @@
 use core::marker::PhantomData;
 
 use embedded_hal::adc::{Channel, OneShot};
+#[cfg(esp32c3)]
+use paste::paste;
 
 #[cfg(esp32c3)]
 use crate::analog::ADC2;
+#[cfg(esp32c6)]
+use crate::clock::clocks_ll::regi2c_write_mask;
+#[cfg(esp32c3)]
+use crate::efuse::Efuse;
+#[cfg(not(esp32c6))]
+use crate::rom::rom_i2c_writeReg_Mask;
 use crate::{
     analog::ADC1,
     peripheral::PeripheralRef,
     peripherals::APB_SARADC,
     system::{Peripheral, PeripheralClockControl},
 };
 
+// constants taken from https://github.com/espressif/esp-idf/blob/045163a2ec99eb3cb7cc69e2763afd145156c4cf/components/soc/esp32s3/include/soc/regi2c_saradc.h
```

Comment:
Please correct the link to point to the c3 header. At least `I2C_SAR_ADC_HOSTID` is different from the linked file